src/Symfony/Component/CssSelector/Tests/XPath/TranslatorTest.php

```diff
@@ -361,6 +367,12 @@ public static function getHtmlIdsTestData()
             [':enabled', ['link-href', 'tag-anchor', 'nofollow-anchor', 'checkbox-unchecked', 'text-checked', 'checkbox-checked', 'area-href']],
             [':disabled', ['checkbox-disabled', 'checkbox-disabled-checked', 'fieldset', 'checkbox-fieldset-disabled']],
             [':checked', ['checkbox-checked', 'checkbox-disabled-checked']],
+            ['div:has(> span)', ['foobar-div']],
+            ['ol:has(> .c)', ['first-ol']],
+            ['div:has( .a)', ['outer-div']],
+            ['a:has(~ a)', ['name-anchor', 'tag-anchor']],
+            ['ol:has(+ p)', ['first-ol']],
+            ['a:has(+ .b)', ['nofollow-anchor']],
```

Comment:
Looks like this is missing a test case for multiple selectors. When `has` has multiple selectors, e.g. `:has(foo, bar)`, it will match if either `foo` **or** `bar` are present.

Reply:
Good catch, I will add these right away.

Reply:
An update for the missing test case @Bilge mentioned above. 

I just saw that @stof has already posted an issue on Cssselect (python parser). Their version does not support relative selector list [here](https://github.com/scrapy/cssselect/issues/138) and remains unresolved until now. I guess this is why this PR stalled.

I will try to find another way.